extensions/auth/opa/src/main/java/org/apache/polaris/extension/auth/opa/OpaAuthorizationConfig.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.extension.auth.opa;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+import java.util.Optional;
+
+/**
+ * Configuration for OPA (Open Policy Agent) authorization.
+ *
+ * <p><strong>Beta Feature:</strong> OPA authorization is currently in Beta and is not a stable
+ * release. It may undergo breaking changes in future versions. Use with caution in production
+ * environments.
+ */
+@ConfigMapping(prefix = "polaris.authorization.opa")
+public interface OpaAuthorizationConfig {
+  Optional<String> url();
+
+  Optional<String> policyPath();
+
+  Optional<AuthenticationConfig> auth();
+
+  Optional<HttpConfig> http();
+
+  /** Validates the complete OPA configuration */
+  default void validate() {
+    checkArgument(url().isPresent() && !url().get().isBlank(), "OPA URL cannot be null or empty");
+    checkArgument(
+        policyPath().isPresent() && !policyPath().get().isBlank(),
+        "OPA policy path cannot be null or empty");
+    checkArgument(auth().isPresent(), "Authentication configuration is required");
+
+    auth().get().validate();
+  }
+
+  /** HTTP client configuration for OPA communication. */
+  interface HttpConfig {
+    @WithDefault("2000")
+    int timeoutMs();
+
+    @WithDefault("true")
```

Comment:
Blindly trusting all presented server certificates is a security risk. Could you add a _severe_ production-readiness (see `ProductionReadinessChecks`) check that fails startup if this is set to `false`?

Reply:
I've thought a bit more about this @snazy and I think using HTTP (and hence without ssl verification) is a legitimate production deployment strategy when we are running the OPA server as a side car to the Polaris server.

Setting `verifySsl=true` and using a `http` url for `policyUrl` will work fine, and it'll just ignore the SSL verification settings. But I'm wondering if we need to be pedantic about what's production ready or not especially when it comes to deducing the security risk for a specific deployment strategy. Let me know what you think!

I've added it for now, along with a warning that OpaPolarisAuthorizer is in beta release, but I'm happy to remove the severe check for `verifySsl=true` if you agree with this.

Reply:
I think those are different scenarios ;)

1. HTTP (no TLS) - no certificate that could be validated.
2. HTTPS (with TLS) - intentionally turn off certificate validation ("not production-ready")
3. HTTPS (with TLS) - with certificate validation, potentially with a custom trust store, recommended for production

With "plaintext" HTTP, which might be, as you mentioned a legit prod deployment strategy, with the known general security drawbacks. It's (hopefully) a well-educated user decision. I'd maybe emit a warning in this case, but would not fight for it.

Scenario 2 is the one that should not work out of the box, as the usage of TLS implies some guarantees. But TLS without certificate & host validation is a pretty bad choice. That's the one that I'm thinking of.

Reply:
Actually, after thinking about this a bit more I'm inclined to leave the `OpaProductionReadinessCheck` as it is. There's a good chance that `policyUri` won't be the only OPA Server Uri we'll be using in the future. Reference: https://trino.io/docs/current/security/opa-access-control.html#batch-mode

Reply:
As long as users can leave the default verify-ssl value for http, that's fine.